### PR TITLE
Feat/keychain support

### PR DESCRIPTION
**Pull request by oed**
WIP of the keychain implementation.

### CODE REVIEW

File: src/three-idx.ts [hunk 1]
```diff
@@ -38,4 +82,92 @@ export class ThreeIDX {
...
+    const newContent = mapEntries.reduce((acc, entry) => Object.assign(acc, entry), content)
+    await this.docs[CDefs.authKeychain].change({ content: newContent })
+  }
+
+  /**
+   * Returns all public keys that is in the auth keychain.
+   */
+  getAllAuthEntries(): Array<AuthEntry> {
+    if (!this.docs[CDefs.authKeychain]) return []
+    const content = Object.assign({}, this.docs[CDefs.authKeychain].content)
```
Comment (PaulLeCam) [inline src/three-idx.ts:152]: It doesn't seem needed to clone the content here as it's not being mutated

File: src/three-idx.ts [hunk 2]
```diff
@@ -38,4 +82,92 @@ export class ThreeIDX {
...
+    return Object.keys(content).map((authLink: string): AuthEntry => content[authLink] as AuthEntry)
+  }
+
+  async pinAllDocs(): Promise<void> {
+    await Promise.all(
+      Object.entries(this.docs).map(async ([_, doc]) => {
```
Comment (PaulLeCam) [inline src/three-idx.ts:158]: Seems this could be replaced by `Object.values(this.docs).map(async (doc) => {` as the key is not used

**Review by PaulLeCam — APPROVED** [review]

**Review by zachferland — APPROVED** [review]
a lot of stuff, nice work, looks good from a read through, but will get a better idea once using in 3id-connect as well 👍

**Comment by oed** [comment]
Released to `next` as `identity-wallet@2.0.0-alpha.2`.

**Pull request merged**